tools/slo-generator/slo_generator/compute.py

```diff
@@ -130,58 +131,68 @@ def make_reports(slo_config,
     Yields:
         list: List of SLO measurement results.
     """
+    slo_full_name = get_full_slo_name(slo_config)
     if backend_method:
         if backend_obj:
             backend_method = getattr(backend_obj, backend_method)
-        LOGGER.info("Backend method: %s (from kwargs).", backend_method)
+        LOGGER.info(
+            f'{slo_full_name} | Backend: {backend_method} (from kwargs).')
     else:
         backend_config = slo_config.get('backend', {})
-        backend_cls = backend_config.get('class')
+        cls = backend_config.get('class')
         method = backend_config.get('method')
-        backend_obj = utils.get_backend_cls(backend_cls)(client=client,
-                                                         **backend_config)
-        backend_method = getattr(backend_obj, method)
-        LOGGER.info("Backend method: %s (from SLO config file).",
-                    backend_cls + '.' + backend_method.__name__)
+        instance = utils.get_backend_cls(cls)(client=client, **backend_config)
+        backend_method = getattr(instance, method)
+        LOGGER.info(f'{slo_full_name :<25} | '
+                    f'Using backend {cls}.{backend_method.__name__} (from '
+                    f'SLO config file).')
 
     # Loop through steps defined in error budget policy and make measurements
     for step in error_budget_policy:
-        good_event_count, bad_event_count = backend_method(
+        backend_result = backend_method(
             timestamp=timestamp,
             window=step['measurement_window_seconds'],
             **slo_config['backend'])
-        report = make_measurement(slo_config, step, good_event_count,
-                                  bad_event_count, timestamp)
-        import pprint
-        pprint.pprint(report)
+        report = make_measurement(slo_config, step, backend_result, timestamp)
         yield report
 
 
-def make_measurement(slo_config, step, good_event_count, bad_event_count,
-                     timestamp):
+def make_measurement(slo_config, step, backend_result, timestamp):
     """Measure following metrics: SLI, SLO, Error Budget, Burn Rate.
 
     Args:
         slo_config (dict): SLO configuration.
         step (dict): Step config.
-        good_event_count (int): Good events count.
-        bad_event_count (int): Bad events count.
+        backend_result (tuple or int): A tuple (good_event_count,
```

Comment:
This method is executed on all SLOs, so it's rather hard to change the interface ATM. The reason I introduced the duality is that for `Prometheus` backend support we have a way of directly computing the SLI in Prometheus itself, but the result is still sent to this master function. 

For the next version we could accept a result class such as this:

```
@dataclass
class BackendResult:
    good_bad_count: tuple = None
    sli_value = None
    
    quantity_on_hand: int = 0

    def __init__(self, name: str, unit_price: float, quantity_on_hand: int=0):
         self.good_bad_count = good_bad_count
         self.sli_value = sli_value
```

and add computation methods to this class instead of the `compute` function.

This is a larger refactoring so I'll open a separate issue for this if that's ok.

Reply:
See https://github.com/ocervell/professional-services/tree/slo-generator-v1.0.0